### PR TITLE
Add support for tasks in Siri

### LINKED ISSUE #5443: Siri. Shortcuts should support tasks as well
Right now shortcuts to start a time entry can have a description, a client, a project and tags. But apparently we missed adding tasks. We should add them too. 

This probably implies a new field in the intent definition file and maybe also changing something in the custom start entry screen.

### PR DESCRIPTION
**Pull request by juanlaube**
## What's this?
Adds support for tasks in all the Siri intents where it makes sense.

### Relationships

Closes #5443 

## Why do we want this?
Tasks!

## How is it done?
Saving the taskId to the intents and handling the actions accordingly.

### Why not another way?
This is the way.

### Side effects
None?

## Review hints
It was impossible for me to test on the simulator or my iPad, for some reason the UI extension shows a loading indicator and doesn't do anything else 🤷‍♂ 

## :squid: Permissions
Only with @Odrakir's blessing.

### CODE REVIEW

File: Toggl.Core.UI/ViewModels/Settings/Siri/SiriShortcutsCustomTimeEntryViewModel.cs [hunk 1]
```diff
@@ -128,33 +128,32 @@ private async Task selectProject()
...
             {
                 return;
             }
 
-            TaskId.Accept(chosenProjectParams.TaskId);
-
             var chosenWorkspace = await interactorFactory.GetWorkspaceById(chosenProjectParams.WorkspaceId).Execute();
 
-            if (!(chosenProjectParams.ProjectId is long chosenProjectProjectId))
+            if (!(chosenProjectParams.ProjectId is long chosenProjectProjectId) || !(chosenProjectParams.TaskId is long chosenProjectTaskId))
```
Comment (Odrakir) [inline Toggl.Core.UI/ViewModels/Settings/Siri/SiriShortcutsCustomTimeEntryViewModel.cs]: And if the shortcut has a project but not task?

File: Toggl.iOS.IntentBinding/ApiDefinition.cs [hunk 2]
```diff
@@ -1,32 +1,43 @@
-﻿using System;
+using System;
 using Foundation;
 using Intents;
 using ObjCRuntime;
 
 namespace Toggl.iOS.Intents
 {
 	// @interface ContinueTimerIntent : INIntent
+	[Watch (5,0), NoTV, NoMac, iOS (12,0)]
```
Comment (Odrakir) [inline Toggl.iOS.IntentBinding/ApiDefinition.cs]: Please, check the [docs](https://github.com/toggl/mobileapp/blob/develop/docs/siri/create-new-intent.md) on how to update intents. You have to remove this line.

Comment (juanlaube) [reply Toggl.iOS.IntentBinding/ApiDefinition.cs]: 🙈 Cleaning these generated files and rebuilding the static library solved the problem, I can't wait to ditch this! 😄

File: Toggl.iOS.IntentStaticLib/Toggl.iOS.IntentStaticLib/StopTimerIntent.m [hunk 3]
```diff
@@ -6,9 +6,9 @@
 
 #import "StopTimerIntent.h"
 
-@implementation StopTimerIntent
-
+#if __has_include(<Intents/Intents.h>) && (!TARGET_OS_OSX || TARGET_OS_IOSMAC) && !TARGET_OS_TV
```
Comment (sofokli1) [inline Toggl.iOS.IntentStaticLib/Toggl.iOS.IntentStaticLib/StopTimerIntent.m]: why are there changes in this file?

Comment (juanlaube) [reply Toggl.iOS.IntentStaticLib/Toggl.iOS.IntentStaticLib/StopTimerIntent.m]: It's something autogenerated by Xcode 11, previously we used Xcode 10.

File: Toggl.iOS.IntentBinding/ApiDefinition.cs [hunk 4]
```diff
@@ -124,7 +124,7 @@ interface ShowReportPeriodIntentResponse
 		[DesignatedInitializer]
 		IntPtr Constructor (ShowReportPeriodIntentResponseCode code, [NullAllowed] NSUserActivity userActivity);
 
-		// +(instancetype _Nonnull)successIntentResponseWithPeriod:(ShowReportPeriodReportPeriod)period;
+		// +(instancetype _Nonnull)successIntentResponseWithPeriod:(ShowReportPeriodReportPeriod)period __attribute__((swift_name("success(period:)")));
```
[the remark this answers is not in the document]
Comment (juanlaube) [reply Toggl.iOS.IntentBinding/ApiDefinition.cs]: Objective Sharpie I guess 🤷‍♂

**Pull request merged**